src/Model/CastleManager.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Created by VisualStudioCode.
+ * User: Thomas
+ * Date: 30/10/2020
+ * Time: 19:00
+ */
+
+namespace App\Model;
+
+use PDO;
+
+class CastleManager extends AbstractManager
+{
+    /**
+     * Class constants
+     */
+    const TABLE = 'castle';
+    const ERROR = -1;
+
+    /**
+     * Class initialization
+     */
+    public function __construct()
+    {
+        parent::__construct(self::TABLE);
+    }
+
+   
+    public function insert(Castle $castle)
+    {
+        // Preparation of the request for the insertion of castle
+        $insert = $this->pdo->prepare("INSERT INTO " . self::TABLE . " (name, score) VALUES (:name, :score)");
+        // Check if access to the database
+        if ((false == $insert)
+            || (false === $insert->bindValue('name', $castle->getName(), PDO::PARAM_STR))
+            || (false === $insert->bindValue('score', $castle->getScore(), PDO::PARAM_INT))) {
+                return self::ERROR;
+        } else {
+            if ($insert->execute()) {
+                return (int)$this->pdo->lastInsertId();
+            }
+        }
+        return "";
+    }
```

Comment:
Your method could probably return only int and therefore have a int type hint.
What's the point of the empty string return at the end ?

Reply:
I deleted the empty string + I deleted "lastInsertId()" and to return directly "$insert->execute()" + the function now returns a boolean and now "error" returns false